daffodil-cli/src/it/scala/org/apache/daffodil/CLI/Util.scala

```diff
@@ -384,6 +385,7 @@ object Util {
 
     def expect(matcher: Matcher[_]): Result = expect.expect(matcher)
     def expect(string: String): Result = expect.expect(contains(string))
+    def expectEOF(): Result = expect.expect(eof())
```

Comment:
If you decide to stop calling expectEOF and getBefore and write to a temp file instead as Steve suggested, then you'll want to revert your changes here.

Reply:
Still should revert this change.

Reply:
Agreed, I think expecting an EOF is usually going to lead to issues, so should be avoided in most cases. If it's not part of the CLI integration "API", it makes it less likely to be used, but it's still possible via `expect.expect(eof)` if it's really needed.

Reply:
Sorry I missed this comment.  done